src/session/session_compact.c

```diff
@@ -334,6 +334,30 @@ __wt_session_compact(WT_SESSION *wt_session, const char *uri, const char *config
     session = (WT_SESSION_IMPL *)wt_session;
     SESSION_API_CALL(session, compact, config, cfg);
 
+    /* Trigger the background server thread and toggle connection statistic. */
+    if ((ret = __wt_config_gets(session, &cfg[1], "background", &cval) == 0)) {
+        if (uri != NULL) {
+            __wt_verbose_warning(session, WT_VERB_COMPACT, "%s",
+              "Background compaction does not work on specific URIs.");
+            goto err;
+        }
+
+        if (cval.val) {
+            /* TODO: Check background compaction server running status and trigger the mutex. */
+        } else {
+            if (__wt_config_gets(session, &cfg[1], "timeout", &cval) == 0 ||
```

Comment:
I see where we are going with this. It might be ok for now but if we had N configuration items, we would need a nice loop here.

Reply:
It would be good if we had something in the API `config_check` that would allow you list incompatible configurations.

Reply:
If we are turning off the `background` compact, do we need to accept the parameters? This code logic should be in the above `if` check.

Reply:
We were thinking that `background=off` should not be given with any other parameters to make sure the application knows what it is doing. Turning the background compaction off and setting other parameters does not make much sense as we are not sure what to do with other parameters.